src/web/src/alert_ack.ecpp

```diff
@@ -67,205 +79,174 @@ bool database_ready;
     }
-    part = zmsg_popstr (recv_msg);
-    if (!part || !streq (part, asset_id.c_str ())) {
-        log_error ("Unexpected reply from '%s'. Expected OK/rule_name/element_name/state. Got OK/rule_name/(null).", mlm_client_sender (client));
-        std::string err =  TRANSLATE_ME ("Bad message.");
+
+    zuuid_t *uuid = zuuid_new ();
+    zmsg_t *request_msg = s_request_ack (uuid, checked_rule_name, checked_element_name, checked_state);
+    rv = client->sendto (AGENT_FTY_ALERT_LIST, RFC_ALERTS_ACK, 1000, &request_msg);
+
+    if (rv == -1)
+    {
+        zuuid_destroy (&uuid);
+        std::string err = TRANSLATE_ME ("client->sendto (address = '%s') failed.", AGENT_FTY_ALERT_LIST);
+        log_fatal (err.c_str ());
         http_die ("internal-error", err.c_str ());
     }
-    part = zmsg_popstr (recv_msg);
-    if (!part || !streq (part, checked_state.c_str ())) {
-        log_error ("Unexpected reply from '%s'. Expected OK/rule_name/element_name/state. Got OK/rule_name/element_name/(null).", mlm_client_sender (client));
-        std::string err =  TRANSLATE_ME ("Bad message.");
+
+    ZmsgGuard resp (client->recv (zuuid_str_canonical (uuid), 5));
+    zuuid_destroy (&uuid);
+    if (!resp)
+    {
+        std::string err = TRANSLATE_ME ("client->recv (timeout = '5') returned NULL");
+        log_fatal (err.c_str ());
         http_die ("internal-error", err.c_str ());
     }
-// TODO: rfc-11 is not written yet, json format can change freely
-</%cpp>
-{
-    <$$ utils::json::jsonify ("rule_name", checked_rule_name) $>,
-    <$$ utils::json::jsonify ("asset_id", asset_id) $>,
-    <$$ utils::json::jsonify ("element_name", checked_element_name) $>,
-    <$$ utils::json::jsonify ("state", checked_state) $>
-}
-<%cpp>
-    return HTTP_OK;
-}
-if (streq (part, "ERROR")) {
-    part = zmsg_popstr (recv_msg);
-    if (!part) {
-        log_error ("Unexpected reply from '%s'. Expected ERROR/reason. Got ERROR/(null).", mlm_client_sender (client));
+
+    if (!streq (client->subject (), RFC_ALERTS_ACK)) {
+        log_error ("Unexpected reply from '%s'. Subject expected = '%s', received = '%s'.",
+            client->sender (), RFC_ALERTS_ACK, client->subject ());
         std::string err =  TRANSLATE_ME ("Bad message.");
         http_die ("internal-error", err.c_str ());
     }
-    if (streq (part, "NOT_FOUND")) {
-        log_error ("Alert identified by rule_name = '%s' and element_name = '%s' does not exist.",
-            checked_rule_name.c_str (), asset_id.c_str ());
-        std::string msg_not_found = TRANSLATE_ME ("Alert identified by rule name = '%s' and element name = '%s'", checked_rule_name.c_str (), checked_element_name.c_str ());
-        http_die ("not-found", msg_not_found.c_str ());
+
+    std::string json;
+    ZstrGuard part (zmsg_popstr (resp));
+    if (streq (part, "CHANGESTATE")) {
+        part = zmsg_popstr (resp);
+        if (!part || !streq (part, zuuid_str_canonical (uuid))) {
```

Comment:
does this test make sense?
uuid was destroyed at line 174

Reply:
Good catch, I've forgotten to merge changes from helper branch.